File: Lab5java/app/src/main/java/com/example/lab5java/FragmentViewPagerAdapter.java
package com.example.lab5java;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class FragmentViewPagerAdapter extends FragmentStateAdapter {
    //初始化
    public FragmentViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 0){
            return new FirstFragment();
        }
        else if (position == 1){
            return new SecondFragment();
        }
        else {
            return new ThirdFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: Lab5kotlin/app/src/main/java/com/example/lab5kotlin/FragmentViewPagerAdapter.kt
package com.example.lab5kotlin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class FragmentViewPagerAdapter  //初始化
    (fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return if (position == 0) {
            FirstFragment()
        } else if (position == 1) {
            SecondFragment()
        } else {
            ThirdFragment()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}